### ISSUE TITLE
add n features to birthyear sex books that are completely random (coin flips)

### ISSUE THREAD

**Issue by msalganik** [varunsatish/llama-recipes-prefer#11]
Let's test if Llama gets confused by adding random features. 

There are many ways to make random features, but the easiest is probably coin flips.  In other words, each column is a binomial random variable.

After adding these columns we need to add them to the book of life.

Note that this will also slightly change book length.

**Comment by keyonvafa** [comment]
My original code for adding extra tokens to the end of input strings might be helpful for this:

https://github.com/keyonvafa/llama-recipes-fertility/blob/main/src/llama_recipes/datasets/fertility_dataset.py#L44

**Comment by msalganik** [comment]
Sorry.  Unclear description.  The things I'm are not padding.  They are noise.  I'm adding n features that are coin flips and seeing if Llama can learn to ignore them.  

It seems important to know if Llama can learn to ignore (which I guess is a form of feature selection).  If not, we have to be careful with what goes in the books.

**Comment by keyonvafa** [comment]
Yep. The padding code I sent gives you one way to do that without changing the underlying data file. The downside is that the same data point would have different coin flips each time, but the upside is that it requires less code. The code you committed is a more complete way to do this that changes the data directly

**Comment by msalganik** [comment]
Interesting.  I guess I didn't understand your code.  To be honest, I still am python illiterate :)

I've added this to our agenda for Wed.  I'd like to understand 1) your code 2) why you wrote it and 3) the trade-offs between writing data files to disk (what I've been doing) and generating data files each time(what you seem to do often).
